litespi/gen.py

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/env python3
+
+#
+# This file is part of LiteSPI.
+#
+# Copyright (c) 2015-2022 Florent Kermarrec <florent@enjoy-digital.fr>
+# Copyright (c) 2022 Victor Suarez Rovere <suarezvictor@gmail.com>
+# Copyright (c) 2020 Xiretza <xiretza@xiretza.xyz>
+# Copyright (c) 2020 Stefan Schrijvers <ximin@ximinity.net>
+# SPDX-License-Identifier: BSD-2-Clause
+
+"""
+LiteSPI standalone core generator
+
+LiteSPI aims to be directly used as a python package when the SoC is created using LiteX. However,
+for some use cases it could be interesting to generate a standalone verilog file of the core:
+- integration of the core in a SoC using a more traditional flow.
+- need to version/package the core.
+- avoid Migen/LiteX dependencies.
+- etc...
+
+The standalone core is generated from a YAML configuration file that allows the user to generate
+easily a custom configuration of the core.
+
+"""
+
+import argparse
+import os
+import yaml
+
+from migen import *
+
+from litex.build.generic_platform import *
+from litex.build.xilinx.platform import XilinxPlatform
+
+from litex.soc.interconnect import wishbone
+from litex.soc.interconnect import axi
+from litex.soc.integration.soc_core import *
+from litex.soc.integration.builder import *
+from litex.soc.integration.soc import SoCRegion
+
+import inspect
+import litespi.modules
+from litespi.spi_nor_flash_module import SpiNorFlashModule
+
+# module database
+modules_dict = {name: cls for name, cls in inspect.getmembers(litespi.modules) if inspect.isclass(cls) and issubclass(cls, SpiNorFlashModule)}
```

Comment:
Can you improve presentation? (not more than 100 characters per line if possible).